Assets/HoloToolkit/Utilities/Editor/ProjectSettingsWindow.cs

```diff
@@ -0,0 +1,274 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+	/// <summary>
+	/// Renders the UI and handles update logic for HoloToolkit/Configure/Apply HoloLens Project Settings.
+	/// </summary>
+	public class ProjectSettingsWindow : AutoConfigureWindow<ProjectSettingsWindow.ProjectSetting>
+	{
+		// Nested Types
+		public enum ProjectSetting
+		{
+			ActiveBuildToWsa,
+			WsaSdkToUwp,
+			WsaUwpBuildToD3D,
+			WsaFastestQuality,
+			WsaEnableVR
+		}
+
+		// Private Methods
+
+		/// <summary>
+		/// Enables virtual reality for WSA and ensures HoloLens is in the supported SDKs.
+		/// </summary>
+		private void EnableVirtualReality()
+		{
+			try
+			{
+				// Grab the text from the project settings asset file
+				string settingsPath = "ProjectSettings/ProjectSettings.asset";
+				string settings = File.ReadAllText(settingsPath);
+
+				// We're looking for the list of VR devices for the current build target, then
+				// ensuring that the HoloLens is in that list
+				bool foundBuildTargetVRSettings = false;
+				bool foundBuildTargetMetro = false;
+				bool foundBuildTargetEnabled = false;
+				bool foundDevices = false;
+				bool foundHoloLens = false;
+
+				var builder = new StringBuilder(); // Used to build the final output
+				string[] lines = settings.Split(new char[] { '\n' });
+				for (int i = 0; i < lines.Length; ++i)
+				{
+					string line = lines[i];
+
+					// Look for the build target VR settings
+					if (!foundBuildTargetVRSettings)
+					{
+						if (line.Contains("m_BuildTargetVRSettings:"))
+						{
+							// If no targets are enabled at all, just create the known entries and skip the rest of the tests
+							if (line.Contains("[]"))
+							{
+								// Remove the empty array symbols
+								line = line.Replace(" []", "\n");
+
+								// Generate the new lines
+								line += "  - m_BuildTarget: Metro\n";
+								line += "    m_Enabled: 1\n";
+								line += "    m_Devices:\n";
+								line += "    - HoloLens";
+
+								// Mark all fields as found so we don't search anymore
+								foundBuildTargetVRSettings = true;
+								foundBuildTargetMetro = true;
+								foundBuildTargetEnabled = true;
+								foundDevices = true;
+								foundHoloLens = true;
+							}
+							else
+							{
+								// The target VR settngs were found but the others
+								// still need to be searched for.
+								foundBuildTargetVRSettings = true;
+							}
+						}
+
+					}
+
+					// Look for the build target for Metro
+					else if (!foundBuildTargetMetro)
+					{
+						if (line.Contains("m_BuildTarget: Metro"))
+						{
+							foundBuildTargetMetro = true;
+						}
+
+					}
+
+					else if (!foundBuildTargetEnabled)
+					{
+						if (line.Contains("m_Enabled"))
+						{
+							line = "    m_Enabled: 1";
```

Comment:
Have these and above strings been verified with Unity 5.5? I remember some of them changing.